File: frontend/src/types/studio.ts
/** LLM configuration extracted from an agent. */
export interface StudioLLMConfig {
  provider: string | null;
  model: string | null;
  temperature: number | null;
  max_tokens: number | null;
}

/** Tool metadata for display. */
export interface StudioToolInfo {
  name: string;
  description: string;
  parameters: Record<string, unknown>;
}

/** Lightweight agent summary for catalog grid. */
export interface StudioAgentSummary {
  agent_id: string;
  name: string;
  display_name: string;
  platform_id: string;
  platform_name: string;
  description: string;
  llm_config: StudioLLMConfig;
  tool_count: number;
  has_system_prompt: boolean;
  agent_class: string;
  status: string;
}

/** Full agent detail including prompt and tools. */
export interface StudioAgentDetail extends StudioAgentSummary {
  system_prompt: string | null;
  tools: StudioToolInfo[];
  parameters: Record<string, unknown>;
}

/** Platform summary with agent counts. */
export interface StudioPlatformSummary {
  id: string;
  name: string;
  description: string;
  version: string;
  agent_count: number;
  agents: string[];
  enabled: boolean;
}

/** Response for GET /studio/agents. */
export interface StudioAgentsListResponse {
  agents: StudioAgentSummary[];
  total: number;
  platforms: StudioPlatformSummary[];
}

/** Response for GET /studio/platforms. */
export interface StudioPlatformsListResponse {
  platforms: StudioPlatformSummary[];
  total: number;
}

File: backend/app/schemas/studio.py
"""
Agent Studio schemas — Rich agent metadata for discovery and catalog.
"""
from typing import Dict, List, Optional, Any
from pydantic import BaseModel


class StudioLLMConfig(BaseModel):
    """LLM configuration extracted from an agent."""
    provider: Optional[str] = None
    model: Optional[str] = None
    temperature: Optional[float] = None
    max_tokens: Optional[int] = None


class StudioToolInfo(BaseModel):
    """Tool metadata for display."""
    name: str
    description: str
    parameters: Dict[str, Any] = {}


class StudioAgentSummary(BaseModel):
    """Lightweight agent summary for catalog grid."""
    agent_id: str               # "{platform_id}/{agent_name}"
    name: str                   # e.g. "research"
    display_name: str           # e.g. "Research Agent"
    platform_id: str            # e.g. "newsletter"
    platform_name: str          # e.g. "Newsletter"
    description: str
    llm_config: StudioLLMConfig
    tool_count: int
    has_system_prompt: bool
    agent_class: str            # e.g. "ResearchAgent"
    status: str = "discovered"


class StudioAgentDetail(StudioAgentSummary):
    """Full agent detail including prompt and tools."""
    system_prompt: Optional[str] = None
    tools: List[StudioToolInfo] = []
    parameters: Dict[str, Any] = {}


class StudioPlatformSummary(BaseModel):
    """Platform summary with agent counts."""
    id: str
    name: str
    description: str
    version: str
    agent_count: int
    agents: List[str]
    enabled: bool


class StudioAgentsListResponse(BaseModel):
    """Response for GET /studio/agents."""
    agents: List[StudioAgentSummary]
    total: int
    platforms: List[StudioPlatformSummary]


class StudioPlatformsListResponse(BaseModel):
    """Response for GET /studio/platforms."""
    platforms: List[StudioPlatformSummary]
    total: int
